lib/sdk/writer/index.js

```diff
@@ -48,29 +102,200 @@ class ImageWriter extends EventEmitter {
     this.pipeline = null
     this.target = null
 
-    this.hadError = false
-
     this.bytesRead = 0
     this.bytesWritten = 0
     this.checksum = {}
   }
 
+  /**
+   * @summary Verify that the selected destination device exists
+   * @param {Function} callback - callback(error)
+   * @private
+   * @example
+   * writer.checkSelectedDevice((error) => {
+   *   // ...
+   * })
+   */
+  checkSelectedDevice (callback) {
+    debug('state:device-select', this.options.path)
+    this.destinationDevice = null
+    drivelist.list((error, drives) => {
+      debug('state:device-select', this.options.path, error ? 'NOT OK' : 'OK')
+
+      if (error) {
+        callback.call(this, error)
+        return
+      }
+
+      const selectedDrive = _.find(drives, {
+        device: this.options.path
+      })
+
+      if (!selectedDrive) {
+        const selectionError = errors.createUserError({
+          title: 'The selected drive was not found',
+          description: `We can't find ${this.options.path} in your system. Did you unplug the drive?`,
+          code: 'EUNPLUGGED'
+        })
+        debug('state:device-select', this.options.path, 'NOT OK')
```

Comment:
Again a small nit, but the 'failure' debug-string here is identical to the one on line 123, so I guess if you see `'state:device-select', this.options.path, 'NOT OK'` (however that happens to get formatted) then I guess it's not immediately obvious whether that's coming from line 123 or line 140?
But I've never used the `debug` module, so maybe this is an invalid concern \*shrug\*